### DISCUSSION TITLE
Tracking search terms

### DISCUSSION THREAD

**Discussion by mhomol** [microsoft-search/pnp-modern-search#2407] [Q&A]
Does the PnP search part support tracking what search terms are being typed into the search box or doing the same type of tracking from the perspective of the Results web part - can that be done via an extension?  If so, can someone point me in the right direction to how this might be done or has been done?

**Comment by FranckyC** [comment]
Hi @mhomol, there is no tracking of any kind right now and there is no way to do this with an extension unfortunately. The only way to do it would be to create your own custom Search box Web Part with your telemetry and connect it to the Search Results WP.

However, I'm thinking to add a 'query modifier' extension at search results Web Part level to do this kind of processing in the future. I'll add it as enhancement,

**Reply by edarroudi** [reply]
@FranckyC you mean like my #2035
I'll do the merge / test very soon. Or did you have something else on your mind?

**Comment by wobba** [comment]
@mhomol I want to provide the opinion of privacy as well - as tracking search terms may violate rules or regulations in your country.
